rest/blip_sync_messages.go

```diff
@@ -0,0 +1,229 @@
+package rest
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"bytes"
+
+	"github.com/couchbase/go-blip"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/channels"
+	"github.com/couchbase/sync_gateway/db"
+)
+
+// Function signature for something that generates a sequence id
+type SequenceIDGenerator func() db.SequenceID
+
+// Helper for handling BLIP subChanges requests.  Supports Stringer() interface to log aspects of the request.
+type subChanges struct {
+	rq                    *blip.Message       // The underlying BLIP message
+	logger                base.SGLogger       // A logger object which might encompass more state (eg, blipContext id)
+	sinceZeroValueCreator SequenceIDGenerator // A sequence generator for creating zero'd since values
+}
+
+// Create a new subChanges helper
+func newSubChanges(rq *blip.Message, logger base.SGLogger, sinceZeroValueCreator SequenceIDGenerator) *subChanges {
+	return &subChanges{
+		rq:                    rq,
+		logger:                logger,
+		sinceZeroValueCreator: sinceZeroValueCreator,
```

Comment:
A callback factory seems heavyweight here - can we just pass in a zeroed SequenceId that can be used as needed?

Reply:
Fixed in https://github.com/couchbase/sync_gateway/pull/3246/commits/5c6d570d7d852ccf297d7c3f1633149604ba629a